In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Libraries loaded successfully!")

Libraries loaded successfully!


In [4]:
# 1. Loading the dataset
df = pd.read_csv('../data/raw/E-commerce_data.csv', encoding='ISO-8859-1')

# 2. Dataset inspection
print("Dataset Shape:", df.shape)
print("\n--- Data Summary ---")
df.info()

# 3. Viewing the first 5 rows of the dataset
df.head()

Dataset Shape: (541909, 8)

--- Data Summary ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  object 
 1   StockCode    541909 non-null  object 
 2   Description  540455 non-null  object 
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  object 
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/10 8:26,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,12/1/10 8:26,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/10 8:26,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/10 8:26,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/10 8:26,3.39,17850.0,United Kingdom


### Initial Data Quality Observations:
* **Missing Values:** `CustomerID` has 135,000 null values (406,829 non-null out of 541,909). `Description` has 1,454 missing values.
* **Data Types:** `InvoiceDate` is currently loaded as `object` (string) and needs to be converted to `datetime64`. `CustomerID` is stored as `float64` due to nulls.
* **Potential Data Issues:** Need to check for negative quantities (cancelled orders starting with 'C' in `InvoiceNo`) and zero/negative `UnitPrice` entries.